src/localStorage.ts

```diff
@@ -0,0 +1,24 @@
+export const loadState = () => {
+  try {
+    const localAllTasksState: string | null = localStorage.getItem(
+      "reduxState"
+    );
+
+    if (!localAllTasksState) return undefined;
+    // when use combinedReducers()
+    // preload state must be a plain object with the same shape as the keys passed to it
+    return { allTasksReducer: JSON.parse(localAllTasksState) };
+  } catch (error) {
+    // if error, return undefined to let reducer initialize app
+    return undefined;
+  }
+};
+
+export const saveState = (reduxState: InitialData) => {
+  try {
+    const localAllTasksState = JSON.stringify(reduxState);
+    localStorage.setItem("reduxState", localAllTasksState);
+  } catch (error) {
+    console.log("Save to local storage error", error);
```

Comment:
Maybe `console.error()` would be more appropriate here?

Reply:
I'm just very used to console.log(error). I think that console.error() is more noticeable??? I guess what is the advantage to use console.error() in this type of situation?

Reply:
I did change it to console.error() btw